pkgs/games/rare/default.nix

```diff
@@ -0,0 +1,63 @@
+{ lib, fetchFromGitHub, buildPythonApplication, makeDesktopItem, copyDesktopItems, qt5
+, pillow, psutil, pypresence, pyqt5, python, qtawesome, requests }:
+
+buildPythonApplication rec {
+  pname = "rare";
+  version = "1.6.2";
+
+  src = fetchFromGitHub {
+    owner = "Dummerle";
+    repo = "Rare";
+    rev = version;
+    sha256 = "sha256-b6PaNpgZXGfn5bzqTXT21eien8w+uPquouqjREqzk1M=";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [
+    copyDesktopItems
+  ];
+
+  propagatedBuildInputs = [
+    pillow
+    psutil
+    pypresence
+    pyqt5
+    qtawesome
+    requests
+  ];
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = pname;
+      exec = "rare";
+      icon = "Rare";
+      comment = meta.description;
+      desktopName = "Rare";
+      genericName = "Rare (Epic Games Launcher Open Source Alternative)";
+    })
+  ];
+
+  preBuild = ''
+    # Solves "PermissionError: [Errno 13] Permission denied: '/homeless-shelter'"
+    export HOME=$(mktemp -d)
+  '';
+
+  postInstall = ''
+    install -Dm644 $out/${python.sitePackages}/rare/resources/images/Rare.png -t $out/share/pixmaps/
+
+    # Solves "qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in "" "
+    wrapProgram $out/bin/rare \
+      --prefix QT_QPA_PLATFORM_PLUGIN_PATH : ${qt5.qtbase.bin}/lib/qt-*/plugins/platforms 
```

Comment:
Sounds like something `wrapQtAppsHook` would solve, check [pkgs/applications/video/corrscope/default.nix](https://github.com/NixOS/nixpkgs/blob/72b193e7c11684e0f0d09f88efcc268b6661f874/pkgs/applications/video/corrscope/default.nix) for an example that integrates with python's wrapping.

Reply:
Yeah, in fact I saw that using that would work, but was not sure how to apply this to a Python project. Going to try it.

Reply:
Hm no, it didn't work. Same issue is showing up:
```
qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
```

Reply:
[0001-Refactor.patch.txt](https://github.com/NixOS/nixpkgs/files/7529515/0001-Refactor.patch.txt)
![Bildschirmfoto_2021-11-12_18-59-20](https://user-images.githubusercontent.com/23431373/141513460-3c1ba0b4-8c15-4d32-b990-b7faaa9d95f7.png)

Reply:
These are the changes I did, besides the change on `all-packages.nix`:

```
diff --git a/pkgs/games/rare/default.nix b/pkgs/games/rare/default.nix
index 6f06b333de9..d56b666fbba 100644
--- a/pkgs/games/rare/default.nix
+++ b/pkgs/games/rare/default.nix
@@ -1,7 +1,6 @@
-{ lib, fetchFromGitHub, buildPythonApplication, makeDesktopItem, copyDesktopItems, qt5
-, pillow, psutil, pypresence, pyqt5, python, qtawesome, requests }:
+{ lib, mkDerivationWith, fetchFromGitHub, python3Packages, wrapQtAppsHook, makeDesktopItem, copyDesktopItems }:

-buildPythonApplication rec {
+mkDerivationWith python3Packages.buildPythonApplication rec {
   pname = "rare";
   version = "1.6.2";

@@ -15,9 +14,10 @@ buildPythonApplication rec {

   nativeBuildInputs = [
     copyDesktopItems
+    wrapQtAppsHook
   ];

-  propagatedBuildInputs = [
+  propagatedBuildInputs = with python3Packages; [
     pillow
     psutil
     pypresence
@@ -43,11 +43,11 @@ buildPythonApplication rec {
   '';

   postInstall = ''
-    install -Dm644 $out/${python.sitePackages}/rare/resources/images/Rare.png -t $out/share/pixmaps/
+    install -Dm644 $out/${python3Packages.python.sitePackages}/rare/resources/images/Rare.png -t $out/share/pixmaps/

-    # Solves "qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in "" "
-    wrapProgram $out/bin/rare \
-      --prefix QT_QPA_PLATFORM_PLUGIN_PATH : ${qt5.qtbase.bin}/lib/qt-*/plugins/platforms
+    ### Solves "qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in "" "
+    ##wrapProgram $out/bin/rare \
+    ##  --prefix QT_QPA_PLATFORM_PLUGIN_PATH : $\{qt5.qtbase.bin}/lib/qt-*/plugins/platforms
   '';

   # Project has no tests
```

Reply:
https://nixos.org/manual/nixpkgs/stable/#qt-runtime-dependencies

> **Note**: `wrapQtAppsHook` ignores files that are non-ELF executables. This means that scripts won’t be automatically wrapped so you’ll need to manually wrap them as previously mentioned. An example of when you’d always need to do this is with Python applications that use PyQt.

`dontWrapQtApps = true;` tells the hook to not bother with wrapping anything on its own, and the `preFixup` code adds the hook's wrapper arguments to the cross-wrapper arguments so the python wrapping hook will pick them up.

Reply:
Ah ok, that's clear.

I did it with another approach, which was, using qt5.wrapQtAppsHook, instead of setting that on the `all-packages.nix` file. The issue was because I found the package on Pypi, so I changed from `fetchFromGitHub` to `fetchPypi` and wasn't able to build it. But besides that, it follows the same idea you specify.